pkg/providers/instance/armutils.go

```diff
@@ -75,6 +75,14 @@ func createVirtualMachineExtension(ctx context.Context, client VirtualMachineExt
 	return &res.VirtualMachineExtension, nil
 }
 
+func GetNic(ctx context.Context, client NetworkInterfacesAPI, rg, nicName string) (*armnetwork.Interface, error) {
```

Comment:
for Gets() it doesnt make as much sense since there is no polling but I just followed the existing pattern. 

For gets it may just make sense to use the client itself directly.